File: Chapter 2-10-UIVIews with Independent Rounded Corners/Objective-C/UIView Independent Corners/UIView+IndependentCorners.m
//
//  UIView+IndependentCorners.m
//  Vacations
//
//  Created by Fireball on 11/07/15.
//  Copyright (c) 2015 Magno Urbano. All rights reserved.
//

#import "UIView+IndependentCorners.h"

@implementation UIView (IndependentCorners)


- (void)setCorners:(UIRectCorner)corners withRadius:(CGFloat)radius
{
  
  // UIButton requires this
  //  [self layer].cornerRadius = 0.0;
  
  UIBezierPath *shapePath = [UIBezierPath bezierPathWithRoundedRect:[self bounds]
                                                  byRoundingCorners:corners
                                                        cornerRadii:CGSizeMake(radius, radius)];
  
  CAShapeLayer *newCornerLayer = [CAShapeLayer layer];
  newCornerLayer.frame = [self bounds];
  newCornerLayer.path = shapePath.CGPath;
  [self layer].mask = newCornerLayer;
  
}


@end

File: Chapter 2-10-UIVIews with Independent Rounded Corners/Swift 2.0/UIView Independent Corners/UIView+IndependentCorners.swift
//
//  UIView+IndependentCorners.swift
//  
//
//  Created by Fireball on 10/07/15.
//
//

import UIKit
import QuartzCore


extension UIView {
  
  
  func corners(corners: UIRectCorner, radius : CGFloat) {
    
    let maskPath = UIBezierPath(roundedRect: bounds, byRoundingCorners: corners, cornerRadii: CGSizeMake(radius, radius))

    let cornerLayer = CAShapeLayer()
    cornerLayer.frame = self.bounds
    cornerLayer.path = maskPath.CGPath
    layer.mask = cornerLayer
    
  }
  
}